src/shell/dotnet/src/Shell/Messaging/MessageRouterStartupAction.cs

```diff
@@ -54,10 +51,18 @@ public Task InvokeAsync(StartupContext startupContext, Func<Task> next)
                             };
                             """));
         }
+        else if (startupContext.ModuleInstance.Manifest.ModuleType == ModuleType.Native)
```

Comment:
I would also consider separating the startupaction to different implementations based on the module type, but I can see that being overcomplicated as well. Give it a thought :)